### ISSUE TITLE
Snowball issues with cross compatibility with Thaumcraft 6

### ISSUE THREAD

**Issue by Seiklus** [Wabbit0101/mods_hoardercraft#40]
Snowballs do not contain the gelum aspect from thaumcraft 6 although they are default minecraft objects when used with this mod. This may be due to an issue with inheriting or overwriting an inheritance in thaumcraft? 

How to do: 
Load vanilla food pantry with thaumcraft and attempt using a thaumcraft crucible with snowballs or snow. I wasn't sure if this was an issue with vanilla food or thaumcraft, so I will post in both issue logs.

**Comment by Wabbit0101** [comment]
VFP doesn't do anything with definitions or recipes for snowballs or snow and it has no explicit integration for thaumcraft. It does have recipes that _use_ snowballs and snow...but why that would affect thaumcraft I don't know. Are there no other mods present?

Don't have any suggestions for you other than perhaps try to isolate to verify it's just these two specific mods. To VFP snowballs are a minecraft object like sticks, with no significance (or changes) to that object.

**Comment by Seiklus** [comment]
I was not able to get the snowball aspects working with this even after isolating the two mods. I am using forge v 14.23.3.2655 (I have also tried the latest release) so I am now assuming it must be on the Thaumcraft side.

**Comment by Dragnier** [comment]
I can confirm this is an issue. Your mod somehow breaks Thaumcraft's Gelum aspect on Snow of all types; blocks, layers, balls. I have no clue why this is the case. Might need to get with Azanor on it somehow. I can't even try to figure out anything because I cannot see your code.

**Comment by Spiderfoxtail** [comment]
> I can confirm this is an issue. Your mod somehow breaks Thaumcraft's Gelum aspect on Snow of all types; blocks, layers, balls. I have no clue why this is the case. Might need to get with Azanor on it somehow. I can't even try to figure out anything because I cannot see your code.

We are having the same issue on our server. Multiple of my players who use Thaumcraft cannot progress further because of the Gelum aspect glitch. We do not wish to remove VFP and hope this glitch can be resolved as soon as possible!

**Comment by Wabbit0101** [comment]
I'll try to look at this to see if I can detect the issue, but honestly (and I do have to code) I have no idea what this is as VFP doesn't actually do anything with snow/snowballs or any form of item substitution or tooltips. But it must be something about how VFP starts up that Thaumcraft does not like (and why just with snow?)

**Comment by Wabbit0101** [comment]
Please see update details at [thaumcraft-beta/issues/1619](https://github.com/Azanor/thaumcraft-beta/issues/1619#issuecomment-460818261).

@Spiderfoxtail Unfortunately because of the current limbo of Thaumcraft and my inability to workaround it (the issue isn't VFP specific and I cannot figure out exactly the recipe "shape" or ingredient context that triggers the stompage) I would say you might have to remove VFP to let your players continue. You could alternately try removing these json recipes see if that helps: `homemade_butter.json` and `cake_sponge.json`

**Comment by Spiderfoxtail** [comment]
It's okay! I rather keep VFP. We have also discovered a work around in our mod pack. We use Fire and Ice which gives Dragon ice that has the lost aspect. So we managed! Thank you for looking into this and please keep up the good work.

**Comment by bookerthegeek** [comment]
Random thought. Can you try renaming `thaumcraft.jar` to `zthaumcraft.jar` and see if that resolves the issue?

**Comment by Wabbit0101** [comment]
Nope. Did not help when I renamed `Thaumcraft-1.12.2-6.1.BETA26.jar` to `zThaumcraft-1.12.2-6.1.BETA26.jar`. Snow-related items still broken.

**Comment by Nikademus** [comment]
Besides the snowball/Gelum problem, the cobweb recipe in the Hedge Alchemy portion of Thaumcraft also breaks when I have VFP installed in combination with BetterWithMods.  If I remove BetterWithMods, the Gelum problem remains.  If I remove VFP, both problems get fixed.

Not sure if this helps troubleshoot or not.

**Comment by sekelsta** [comment]
Thaumcraft + VFP (and no other mods, except baubles because thaumcraft needs it) makes rotten flesh give metallum instead of the usual things for some reason, also no aspects for snowballs, but Thaumcraft on its own works fine.

**Comment by The-Arcanian** [comment]
I don't have metallum on rotten flesh, but I did have the problem with gelum. I think it has to do with Thaumcraft's recursive aspect attribution from crafting. Because of VFP's crafting recipe for snowballs they and everything crafted from them lose their aspects. I was able to work around it using Mod Tweaker to add the aspects back.
